tests/stack_integrity_thread.cc

```diff
@@ -8,6 +8,26 @@ using namespace CHERI;
 
 bool *threadStackTestFailed;
 
+/*
+ * Define a macro that gets a __cheri_callback capability and calls it, while
+ * support adding instruction before the call. This is used to avoid code
+ * duplication, in cases we want to call a __cheri_callback in multiple
+ * places while adding additional functionalities.
+ */
+#define CALL_CHERI_CALLBACK(fn, instruction, additional_input)                 \
```

Comment:
Very Nit: `fn` should probably have a better name like `handle` because it is not really a function.

Reply:
It also might be less confusing to say that `additional_input` is referenced in the `instruction`. I got really confused when I saw `additional_input` not referenced anywhere in the macro.

Reply:
Great note, thanks @JerryHsia-MS !